ipn/ipnlocal/local_test.go

```diff
@@ -1330,3 +1331,317 @@ func (rc *routeCollector) AdvertiseRoute(pfx netip.Prefix) error {
 	rc.routes = append(rc.routes, pfx)
 	return nil
 }
+
+func setSyspolicyHandlerForTest(tb testing.TB, h syspolicy.Handler) {
+	tb.Helper()
+	oldHandler := syspolicy.PolicyHandler
+	syspolicy.PolicyHandler = h
+	tb.Cleanup(func() {
+		syspolicy.PolicyHandler = oldHandler
+	})
+}
+
+type mockSyspolicyHandler struct {
+	t          *testing.T
+	key        syspolicy.Key
+	s          string
+	exitNodeID string
+	exitNodeIP string
+	bothKeys   bool
+	err        error
+}
+
+func (h *mockSyspolicyHandler) ReadString(key string) (string, error) {
+	if key != string(h.key) {
+		return "", h.err
+	}
+	if h.bothKeys { // when both keys are used
+		switch key {
+		case string(syspolicy.ExitNodeID):
+			return h.exitNodeID, h.err
+		case string(syspolicy.ExitNodeIP):
+			return h.exitNodeIP, h.err
+		}
+	}
+	return h.s, h.err
+}
+
+func (h *mockSyspolicyHandler) ReadUInt64(key string) (uint64, error) {
+	return 0, h.err
+}
+
+func (h *mockSyspolicyHandler) ReadBoolean(key string) (bool, error) {
+	return false, h.err
+}
+
+func TestSetExitNodeIDPolicy(t *testing.T) {
+	pfx := netip.MustParsePrefix
+	tests := []struct {
+		name           string
+		key            syspolicy.Key
+		exitNodeID     string
+		exitNodeIP     string
+		prefs          *ipn.Prefs
+		exitNodeIPWant string
+		exitNodeIDWant string
+		prefsChanged   bool
+		nm             *netmap.NetworkMap
+		bothKeys       bool //indicates if we want to handle both ExitNodeID and ExitNodeIP keys in a test.
```

Comment:
The relationship between `key` and `bothKeys` is complicated and kind of redundant. I recommend instead having a pair of booleans to say if each one is in use. Or treat each one as in use if `exitNodeXX != ""`

Either way, I think getting rid of `key` and `bothKeys` is better because they're kind of confusing right now.